Framework/Muon/test/ApplyMuonDetectorGroupingTest.h

```diff
@@ -0,0 +1,777 @@
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("SubtractedPeriods", emptyString));
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+
+    TS_ASSERT(wsGroup->getItem("inputGroup; Group; test; Counts; #1"));
+    TS_ASSERT(wsGroup->getItem("inputGroup; Group; test; Counts; #1_Raw"));
+
+    auto wsOut = wsGroup->getItem("inputGroup; Group; test; Counts; #1_Raw");
+
+    CompareWorkspaces algCompare;
+    TS_ASSERT_THROWS_NOTHING(algCompare.initialize());
+    TS_ASSERT(algCompare.isInitialized());
+    TS_ASSERT_THROWS_NOTHING(
+        algCompare.setProperty("Workspace1", ws->getName()));
+    TS_ASSERT_THROWS_NOTHING(
+        algCompare.setProperty("Workspace2", wsOut->getName()));
+    TS_ASSERT_THROWS_NOTHING(algCompare.setProperty("Tolerance", 0.001));
+    TS_ASSERT_THROWS_NOTHING(algCompare.setProperty("CheckAllData", true));
+    TS_ASSERT_THROWS_NOTHING(algCompare.execute());
+    TS_ASSERT(algCompare.isExecuted());
+
+    TS_ASSERT(algCompare.getProperty("Result"));
+
+    AnalysisDataService::Instance().clear();
+  }
+
+  void test_groupingWithCounts() {
+    std::string emptyString("");
+
+    auto ws = createCountsWorkspace(5, 10, 0.0);
+    auto wsGroup = boost::make_shared<WorkspaceGroup>();
+    AnalysisDataService::Instance().addOrReplace("inputData", ws);
+    AnalysisDataService::Instance().addOrReplace("inputGroup", wsGroup);
+
+    std::string wsName = ws->getName();
+    std::string wsGroupName = (wsGroup)->getName();
+
+    TS_ASSERT_EQUALS(wsGroup->getNumberOfEntries(), 0);
+    TS_ASSERT_EQUALS(ws->getNumberHistograms(), 5);
+
+    ApplyMuonDetectorGrouping alg;
+    TS_ASSERT_THROWS_NOTHING(alg.initialize());
+    TS_ASSERT(alg.isInitialized());
+
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("InputWorkspace", ws->getName()));
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("InputWorkspaceGroup", wsGroup->getName()));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("groupName", "test"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("Grouping", "1,2,3"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("AnalysisType", "Counts"));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeMin", 0.0));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeMax", 30.0));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("RebinArgs", emptyString));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("TimeOffset", 0.0));
+    TS_ASSERT_THROWS_NOTHING(
+        alg.setProperty("SummedPeriods", std::to_string(1)));
+    TS_ASSERT_THROWS_NOTHING(alg.setProperty("SubtractedPeriods", emptyString));
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+
+    TS_ASSERT(wsGroup->getItem("inputGroup; Group; test; Counts; #1"));
+    TS_ASSERT(wsGroup->getItem("inputGroup; Group; test; Counts; #1_Raw"));
+
+    auto wsOut = boost::dynamic_pointer_cast<MatrixWorkspace>(
+        wsGroup->getItem("inputGroup; Group; test; Counts; #1_Raw"));
+
+    TS_ASSERT_DELTA(wsOut->readX(0)[0], 0.000, 0.001);
```

Comment:
in future you could define a variable for the delta

Reply:
In this case that's quite difficult as I'm using a delta which is 3 sig figs compared to the expected value (which, depending on the axis, varies by several orders of magnitude). So it was just quicker to hard code them.